web_responsive/static/src/js/web_responsive.js

```diff
@@ -358,7 +405,61 @@ odoo.define('web_responsive', function (require) {
             } else {
                 this._super.apply(this, arguments);
             }
-        }
+        },
+    });
+
+    // Hide AppMenu & remove waiting anim when loaded action
+    ActionManager.include({
+        doAction: function (action, options) {
+            return this._super.apply(this, arguments).always(function () {
+                if (options && 'action_menu_id' in options) {
+                    // Need hide AppMenu?
+                    var $app_menu = $('.o_menu_apps .dropdown');
+                    if (!$app_menu.length) {
+                        return;
+                    }
+                    var $app = $app_menu.find('.o_app.active');
+                    var app_menu_id = $app.data('menuId');
+                    if (app_menu_id === options.action_menu_id) {
+                        // Now uses this way because Odoo 12.0 uses
+                        // bootstrap 4.1
+                        // TODO: Change to use "hide" method in modern
+                        // bootstrap versions (>4.1)
+                        if ($app_menu.hasClass('show')) {
+                            $app_menu.dropdown('toggle');
+                        }
+
+                        $(document.body).removeClass('o-cursor-waiting');
+                        $app_menu.find('.o-app-waiting').removeClass(
+                            'o-app-waiting');
+                    }
+                } else if (action) {
+                    // Need hide Menu?
+                    var $menu = $('.o_menu_sections');
+                    if (!$menu.length) {
+                        return;
+                    }
+                    var $item = $menu.find('.o-menu-item-waiting');
+                    if (!$item.length) {
+                        return;
+                    }
+                    var action_id = $item.data('actionId');
+                    if (action_id === action.id) {
+                        $item.removeClass('o-menu-item-waiting');
+
+                        $menu.find('li.show').each(
+                            function (i, el) {
+                                if ($.contains(el, $item[0])) {
+                                    $(el).find('.dropdown-toggle')
+                                        .dropdown('toggle');
+                                }
+                            });
+
+                        $(document.body).removeClass('o-cursor-waiting');
```

Comment:
Both of these calls are inside `if` blocks.

If none of them evaluate to true, nothing would remove this class. Is that the expected behavior, or should this call be moved to the end, outside of both `if`s?

I cannot test because there's no runbot... :confused:

Reply:
@Tardo please confirm this

Reply:
Added a "fallback", waiting cursor are removed after ~3 seconds (is the time it takes to appear the BlockUI)